attachment_zipped_download/models/ir_attachment.py

```diff
@@ -28,9 +28,18 @@ def _create_temp_zip(self):
         zip_buffer = BytesIO()
         with zipfile.ZipFile(zip_buffer, "a", zipfile.ZIP_DEFLATED, False) as zip_file:
             for attachment in self:
-                zip_file.write(
-                    attachment._full_path(attachment.store_fname), attachment.name
+                # Use attachment read for two purpose (initial implementation wasn't):
+                # * to ensure access write
+                # * do not assume file is stored on filesystem
+                zip_file.writestr(
+                    attachment._compute_zip_file_name(),
+                    attachment.read(["raw"])[0]["raw"],
```

Comment:
Not sure `read` is better than `attachment.datas`

Reply:
I'll give a try to use `datas` instead of `raw`. Also I'll re-try to direct access to the record attribute but I believes that the added test will failed and we wont call the check('read') ...

Reply:
I confirm that test will faill with `attachment.datas` so I can either use:
`attachement.check("read")` before or in one line `attachment.read(["datas"])[0]["datas"]`. Do you have any preferences,  explicity calling attachement.check("read") seems probably easier to read and understand :thinking: 

I'll retry to various type of files to make sure it works with binary and text files...

Reply:
Yes, it's better to call it explicitly for clarity and for reusing cache. I think it's not called because it's already in cache.

Reply:
done ! And I confirm it's better to use `raw` because `datas` convert content to base64 which is not desired before zipping file !

Reply:
I've test with various format: .jpeg, .png, .pdf, .md, .cfg, .xlsx and were able unzip and read files afterwards